File: web-react/src/models/recipe.ts
import { Category } from "./category";

export class Recipe {
    public id: number;
    public name: string;
    public category: Category | undefined;
    public ingredients: Array<string>;
    public instructions: Array<string>;
    constructor(id: number, name: string, category: Category, ingredients: Array<string>, instructions: Array<string>) {
        this.id = id;
        this.name = name;
        this.category = category;
        this.ingredients = ingredients;
        this.instructions = instructions;
    }
}

File: api-dotnet/Cookbook/Cookbook/Recipe.cs
namespace Cookbook;

public class Recipe
{
    public required long Id { get; set; }
    public required string Name { get; set; }
    public required string Category { get; set; }
    public required IEnumerable<string> Ingredients { get; set; }
    public required IEnumerable<string> Instructions { get; set; }
}
